docs/standard/microservices-architecture/microservice-ddd-cqrs-patterns/nosql-database-persistence-infrastructure.md

```diff
@@ -119,16 +116,219 @@ You can see that the way you work with your domain model can be similar to the w
+                  new GeoJsonLinearRingCoordinates<GeoJson2DGeographicCoordinates>(
+                                                                 coordinatesList)));
+    }
+}
+```
+
+You can see there are a few attributes and types coming from the MongoDB NuGet packages.
+
+NoSQL databases are usually very well suited to work with non-relational data but hierarchical. In this example, we are using MongoDB types expecially made for geo-locations, like `GeoJson2DGeographicCoordinates`.
+
+#### Retrieve the database and the collection
+
+In eShopOnContainers we have created a custom database context where we implement the code to retrieve the database and the MongoCollections, as in the following code.
+
+```csharp
+public class LocationsContext
+{
+    private readonly IMongoDatabase _database = null;
+
+    public LocationsContext(IOptions<LocationSettings> settings)
+    {
+        var client = new MongoClient(settings.Value.ConnectionString);
+        if (client != null)
+            _database = client.GetDatabase(settings.Value.Database);
+    }
+
+    public IMongoCollection<Locations> Locations
+    {
+        get
+        {
+            return _database.GetCollection<Locations>("Locations");
+        }
+    }       
+}
+```
+
+#### Retrieve the data
+
+Then, from any C# code, like Web API controllers or custom Repositories, you can write similar code to the following when querying through the MongoDB API, being the `_context` object an instance of the previous `LocationsContext` class.
+
+```csharp
+public async Task<Locations> GetAsync(int locationId)
+{
+    var filter = Builders<Locations>.Filter.Eq("LocationId", locationId);
+    return await _context.Locations
+                            .Find(filter)
+                            .FirstOrDefaultAsync();
+}
+```
+
+#### Use an env-var in the docker-compose.override.yml file for the MongoDB connection string
+
+When creating a MongoClient object, the needed parameter is precisely the ConnectionString pointing, in the case of eShopOnContainers, either to a local MongoDB Docker container or to the “production” Azure Cosmos DB database. That connection string comes from the envitronment variables defined at the docker-compose.override.yml files when deploying with docker-compose or Visual Studio, as in the following yml code.
+
+```yml
+# docker-compose.override.yml
+version: '3'
+services:
+  # Other services
+  locations.api:
+    environment:
+      # Other settings
+      - ConnectionString=${ESHOP_AZURE_COSMOSDB:-mongodb://nosql.data}
+
+```
+
+The `ConnectionString` environment vartiable is defined so if the `ESHOP_AZURE_COSMOSDB` global variable is defined at the `.env` file with the Azure Cosmos DB connection string, it will it to access the Azure Cosmos DB database in the cloud. 
+
+Here you can see the `.env` file with the Azure Cosmos DB connection string gloval environment variable, used at eShopOnContainers as a convenient approach.
```

Comment:
gloval --> global
no comma after variable
Better: The following shows the `.env` file with the Azure Cosmos DB connection string global environment variable:

Reply:
👍